src/backend/distributed/utils/distribution_column.c

```diff
@@ -129,6 +132,12 @@ BuildDistributionKeyFromColumnName(Relation distributedRelation, char *columnNam
 	Var *distributionColumn = NULL;
 	char *tableName = RelationGetRelationName(distributedRelation);
 
+	/* short circuit for reference tables */
```

Comment:
Just a note for the reviewer. I see that this is a hacky solution. However, my other attempts to refactor the function lead to many other cascading changes.The main reason is that once the function is called, `distributedRelation` is not actually distributed. So, it is not possible to learn the distribution method. Are you OK with that?